### DISCUSSION TITLE
Get the output when there's an error?

### DISCUSSION THREAD

**Discussion by gak** [rust-shell-script/rust_cmd_lib#29] [Q&A] [answered]
Hi!

Is it possible to fetch the output (std and err) from when command fails?

Thanks :)

**Comment by rust-shell-script** [comment, marked as answer]
My original thought was that if command failed, then the output was not reliable anymore, so it was better to just return error. Do you have a concrete example about such usage ?

I guess you can use `spawn!(xxx 2>&1)?.wait_with_pipe(&mut |pipe| {...})` for now. In the future, I could add a builtin command `ignore` for such cases, similar to rustup install script.

**Reply by gak** [reply]
My usage is I'm running tests against a binary to see how well the CLI acts, and I want to know the output so that I can let the user know why the test failed.

I ended up using logging and forcing the level to `info`.

If I want to try something a bit nicer I'll try out the `spawn!()` chain.

If you're curious, the cli test lives in https://github.com/feeless/feeless/tree/main/examples/cli where `main.rs` has the test bootstrap, and the other modules have the tests that run things.

Thanks for the awesome crate. Keep up the good work! :)
